### PR TITLE
Revives PR #58579; Sligh refactor to AI datums that allows for basic support of subtrees

### PR DESCRIPTION
**Pull request by ma44**
## About The Pull Request
Revival of #58579

## Why It's Good For The Game
Subtyped controllers is practically required to make simple mobs datumized without needing to redesign the entire system to work in AI datum's current configuration.

### CODE REVIEW

**Comment by ma44** [comment]
Needs some more documentation alongside another look at the subtyped behaviors to ensure they're equivalent to master branch.

**Comment by ike709** [comment]
Based

**Comment by tralezab** [comment]
dopeness!

**Comment by ma44** [comment]
Going to need some conversion to LAZYLIST to address previous PR's comments alongside a relook at SUBTREE_RETURN_FINISH_PLANNING to have it's documentation actually function as it should.

File: code/datums/ai/_ai_controller.dm [hunk 1]
```diff
@@ -144,7 +173,15 @@ have ways of interacting with a specific atom and control it. They posses a blac
...
 /datum/ai_controller/proc/SelectBehaviors(delta_time)
 	SHOULD_NOT_SLEEP(TRUE) //Fuck you don't sleep in procs like this.
-	return
+	if(!COOLDOWN_FINISHED(src, failed_planning_cooldown))
+		return FALSE
+
+	LAZYINITLIST(current_behaviors)
+
+	if(LAZYLEN(planning_subtrees))
+		for(var/datum/ai_planning_subtree/subtree in planning_subtrees)
```
Comment (Kylerace) [inline code/datums/ai/_ai_controller.dm:182]: should be as anything

**Review by Kylerace — CHANGES_REQUESTED** [review]

**Pull request merged**